sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,645 @@
+ *     puts("Error creating DTLS sock");
+ *     return -1;
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * Now we can listen to incoming packets using @ref sock_dtls_recv(). The
+ * application waits indefinitely for new packets. If we want to timeout this
+ * waiting period we could alternatively set the `timeout` parameter of the
+ * function to a value != @ref SOCK_NO_TIMEOUT. If an error occurs we just
+ * ignore it and continue looping. We can reply to an incoming message using
+ * its `session`.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * while (1) {
+ *     int res;
+ *     char buf[128];
+ *     sock_dtls_session_t session;
+ *
+ *     res = sock_dtls_recv(&dtls_sock, &session, buf, sizeof(buf),
+ *                          SOCK_NO_TIMEOUT);
+ *     if (res > 0) {
+ *         printf("Received %d bytes: %s -- echo!\n", res, buf);
+ *         if (sock_dtls_send(&dtls_sock, &session, buf, res) < 0) {
+ *             puts("Error sending reply");
+ *         }
+ *     }
+ * }
+ * return 0;
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * ### Client Operation
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include "net/sock/udp.h"
+ * #include "net/sock/dtls.h"
+ * #include "net/ipv6/addr.h"
+ * #include "net/credman.h"
+ *
+ * #define SOCK_DTLS_CLIENT_TAG (20)
+ *
+ * #ifndef SERVER_ADDR
+ * #define SERVER_ADDR "fe80::aa:bb:cc:dd" // replace this with the server address
+ * #endif
+ *
+ * int main(void)
+ * {
+  *    // Add credentials
+ *     // [...]
+ *
+ *     // initialize client
+ *     char rcv[128];
+ *     sock_udp_t udp_sock;
+ *     sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ *     local.port = 12345;
+ *
+ *     sock_udp_ep_t remote;
+ *     remote.port = DTLS_DEFAULT_PORT;
+ *     remote.netif = gnrc_netif_iter(NULL)->pid;   // only if GNRC_NETIF_NUMOF == 1
+ *
+ *     sock_dtls_t dtls_sock;
+ *     sock_dtls_session_t session;
+ *
+ *     if (!ipv6_addr_from_str((ipv6_addr_t *)remote.addr.ipv6, SERVER_ADDR)) {
+ *         puts("Error parsing destination address");
+ *         return -1;
+ *     }
+ *
+ *     if (sock_udp_create(&udp_sock, &local, NULL, 0) < 0) {
+ *         puts("Error creating UDP sock");
+ *         return -1;
+ *     }
+ *
+ *     if (sock_dtls_create(&dtls_sock, &udp_sock,
+ *                          SOCK_DTLS_CLIENT_TAG,
+ *                          SOCK_DTLS_1_2, SOCK_DTLS_CLIENT) < 0) {
+ *         puts("Error creating DTLS sock");
+ *         sock_udp_close(&udp_sock);
+ *         return -1;
+ *     }
+ *
+ *     if (sock_dtls_session_create(&dtls_sock, &remote, &session) < 0) {
+ *         puts("Error creating session");
+ *         sock_dtls_close(&dtls_sock);
+ *         sock_udp_close(&udp_sock);
+ *         return -1;
+ *     }
+ *
+ *     const char data[] = "HELLO";
+ *     int res = sock_dtls_send(&dtls_sock, &session, data, sizeof(data));
+ *     if (res >= 0) {
+ *         printf("Sent %d bytes: %*.s\n", res, res, data);
```

Comment:
Here as well.

Reply:
I guess you meant line 355 instead? This line only prints `data` which is a `\0`-terminated string.